Scripts/script-StringReplace.yml

```diff
@@ -0,0 +1,78 @@
+commonfields:
+  id: StringReplace
+  version: -1
+name: StringReplace
+script: |-
+  if ((typeof args.data) !== 'string') {
+      return {
+          Type: entryTypes.error,
+          ContentsFormat: formats.text,
+          Contents: 'Invalid input - please make sure \'data\' is of type string'
+      }
+  }
+
+  var arr = [];
+  if (args.replaceAll === 'true') {
+      arr.push('g');
+  }
+  if (args.caseInsensitive === 'true') {
+      arr.push('i');
+  }
+  if (args.multiLine === 'true') {
+      arr.push('m');
+  }
+
+  var newValue = args.newValue === "''" ? "" : args.newValue;
+  var r = new RegExp(args.regex, arr.join(''));
+  var val = args.data.replace(r, newValue);
+
+  //if no match found - return the original string
+  setContext('StringReplace.Result', val);
```

Comment:
Using SetContext is risky. Encountered several time a race condition with it. I would just return it as output for the next tasks.

Reply:
We already think about it. Just return the result is not enough. We want to have all the script some output so the users can easily use that as input. But you right about race condition, but in other hand if we use `EntryContext` instead, then at the first run it will set output to `StringReplace.Result` and at the second run it will make it array, which is not user friendly (user must modify the output from source). This is an open issue, something that we need to decide how we want to make it user friendly and usable